### PR TITLE
Issue a nicer error message on wasm components.

### PR DESCRIPTION
**Pull request by sunfishcode**
Decode just enough of the component binary format to recognize when the input is a component, and issue a dedicated error message for it.

Before:

0000008: error: bad wasm file version: 0x1000d (expected 0x1)

After:

0000008: error: wasm components are not yet supported in this tool

### CODE REVIEW

**Review by SoniEx2 — APPROVED** [review]

**Comment by SoniEx2** [comment]
this should probably have a dedicated test for it fwiw

File: include/wabt/binary.h [hunk 1]
```diff
@@ -21,6 +21,8 @@
 
 #define WABT_BINARY_MAGIC 0x6d736100
 #define WABT_BINARY_VERSION 1
+#define WABT_MODULE_LAYER 0
+#define WABT_COMPONENT_LAYER 1
```
Comment (sbc100) [inline include/wabt/binary.h:25]: These should probably have `WASM_BINARY_` prefix like everything else here.

How about `WASM_BINARY_LAYER_MODULE` and `WASM_BINARY_LAYER_COMPONENT`?

Comment (sunfishcode) [reply include/wabt/binary.h]: Done.

**Comment by sunfishcode** [comment]
@sbc100 I've addressed the review feedback; would you mind taking a look? Thanks!

**Review by sbc100 — APPROVED** [review]

**Comment by sunfishcode** [comment]
Rebased on main to pick up the CI fixes.

**Pull request merged**